jobs/vqsr.py

```diff
@@ -0,0 +1,1033 @@
+           might not be yet generated the point of Batch job submission
+    @param intervals_path: path to specific interval list
+    @param out_path: path to write final recalibrated VCF to
+    @param use_as_annotations: use allele-specific annotation for VQSR
+    @param overwrite: whether to not reuse intermediate files
+    @param job_attrs: default job attributes
+    @return: a final Job, and a path to the VCF with VQSR annotations
+    """
+    resources = {
+        key: b.read_input_group(
+            base=str(reference_path(f'broad/{key}_vcf')),
+            index=str(reference_path(f'broad/{key}_vcf_index')),
+        )
+        for key in [
+            'dbsnp',
+            'hapmap',
+            'omni',
+            'one_thousand_genomes',
+            'mills',
+            'axiom_poly',
+        ]
+    }
+
+    is_small_callset = gvcf_count < 1000
+    # For small callsets, we gather the VCF shards and collect QC metrics directly.
+    # For anything larger, we need to keep the VCF sharded and gather metrics
+    # collected from them.
+    is_huge_callset = gvcf_count >= 100000
+    # For huge callsets, we allocate more memory for the SNPs Create Model step
+
+    # To fit only a site-only VCF
+    small_disk = 50 if is_small_callset else (100 if not is_huge_callset else 200)
+    # To fit a joint-called VCF
+    medium_disk = 100 if is_small_callset else (200 if not is_huge_callset else 500)
+    huge_disk = 200 if is_small_callset else (500 if not is_huge_callset else 2000)
+
+    if get_config()['workflow']['sequencing_type'] == 'genome':
+        scatter_count = gvcf_count // 10
+    else:
+        assert get_config()['workflow']['sequencing_type'] == 'exome'
+        scatter_count = gvcf_count // 1000
+
+    jobs: list[Job] = []
+
+    if input_vcf_or_mt_path.name.endswith('.mt'):
+        # Importing dynamically to make sure $CPG_DATASET_GCP_PROJECT is set.
+        from analysis_runner import dataproc
+
+        assert meta_ht_path
+        assert hard_filter_ht_path
+        job_name = 'VQSR: MT to site-only VCF'
+        combined_vcf_path = tmp_prefix / 'input.vcf.gz'
+        if not can_reuse(combined_vcf_path, overwrite):
+            mt_to_vcf_job = dataproc.hail_dataproc_job(
+                b,
+                f'jobs/dataproc_scripts/mt_to_siteonlyvcf.py --overwrite '
+                f'--mt {input_vcf_or_mt_path} '
+                f'--meta-ht {meta_ht_path} '
+                f'--hard-filtered-samples-ht {hard_filter_ht_path} '
+                f'-o {combined_vcf_path} ',
+                max_age='8h',
+                packages=[
+                    'cpg_utils',
+                    'cpg_gnomad',  # github.com/populationgenomics/gnomad_methods
+                    'click',
+                    'google',
+                    'fsspec',
+                    'sklearn',
+                    'gcloud',
+                    'selenium',
```

Comment:
Do we need all these?

Reply:
We don't actually run VQSR from a matrix table in this repo (that's part of the large-cohort pipeline), so removed calling that script.